hpo_demo_setup.sh

```diff
@@ -0,0 +1,289 @@
+## TODO: Get the version from hpo repo
+HPO_VERSION="0.01"
+
+# Default cluster
+CLUSTER_TYPE="native"
+PY_CMD="python3.6"
+LOGFILE="${PWD}/hpo.log"
+export N_TRIALS=3
+export N_JOBS=1
+
+function usage() {
+	echo "Usage: $0 [-s|-t] [-d] [-i hpo-image] [-r] [-c cluster-type]"
+	echo "s = start (default), t = terminate"
+	echo "r = restart hpo only"
+	echo "d = Don't start experiments"
+	echo "p = expose prometheus port"
+	exit 1
+}
+
+# get date in format
+function get_date() {
+	date "+%Y-%m-%d %H:%M:%S"
+}
+
+function time_diff() {
+	ssec=`date --utc --date "$1" +%s`
+	esec=`date --utc --date "$2" +%s`
+
+	diffsec=$(($esec-$ssec))
+	echo $diffsec
+}
+
+function check_err() {
+	err=$?
+	if [ ${err} -ne 0 ]; then
+		echo "$*"
+		exit 1
+	fi
+}
+
+###########################################
+#   Clone HPO git Repos
+###########################################
+function clone_repos() {
+	echo
+	echo "#######################################"
+	echo " Cloning hpo git repos"
+	if [ ! -d hpo ]; then
+		git clone git@github.com:kruize/hpo.git 2>/dev/null
+		if [ $? -ne 0 ]; then
+			git clone https://github.com/kruize/hpo.git 2>/dev/null
+		fi
+		check_err "ERROR: git clone of kruize/hpo failed."
+	fi
+
+	if [ ! -d benchmarks ]; then
+		git clone git@github.com:kruize/benchmarks.git 2>/dev/null
+		if [ $? -ne 0 ]; then
+			git clone https://github.com/kruize/benchmarks.git 2>/dev/null
+		fi
+		check_err "ERROR: git clone of kruize/benchmarks failed."
+	fi
+	echo "done"
+	echo "#######################################"
+	echo
+}
+
+###########################################
+#   Cleanup HPO git Repos
+###########################################
+function delete_repos() {
+	echo " Delete hpo and benchmarks git repos"
+	rm -rf hpo benchmarks
+}
+
+###########################################
+#   Start HPO
+###########################################
+function hpo_install() {
+	echo
+	echo "#######################################"
+	echo " Start HPO Server"
+	if [ ! -d hpo ]; then
+		echo "ERROR: hpo dir not found."
+		if [ ${hpo_restart} -eq 1 ]; then
+			echo "ERROR: HPO not running. Wrong use of restart command"
+		fi
+		exit -1
+	fi
+	pushd hpo >/dev/null
+		if [ -z "${HPO_DOCKER_IMAGE}" ]; then
+			HPO_DOCKER_IMAGE=${HPO_DOCKER_REPO}:${HPO_VERSION}
+		fi
+		if [[ ${CLUSTER_TYPE} == "native" ]]; then
+			echo
+			echo "Starting hpo with  ./deploy_hpo.sh -c ${CLUSTER_TYPE}"
+			echo
+			./deploy_hpo.sh -c ${CLUSTER_TYPE} >> ${LOGFILE}  &
+			#check_err "ERROR: HPO failed to start, exiting"
+		else
+			echo
+			echo "Starting hpo with  ./deploy_hpo.sh -c ${CLUSTER_TYPE} -h ${HPO_DOCKER_IMAGE}"
+			echo
+
+			./deploy_hpo.sh -c "${CLUSTER_TYPE}" -h "${HPO_DOCKER_IMAGE}"
+			#check_err "ERROR: HPO failed to start, exiting"
+		fi
+	popd >/dev/null
+	echo "#######################################"
+	echo
+}
+
+###########################################
+#   Start HPO Experiments
+###########################################
```

Comment:
Please add more comments here to indicate how this can be customized for other usecases